packages/signers/src/signers/utils/index.ts

```diff
@@ -213,6 +214,49 @@ export async function getAALastAuthenticatorId(
   return data.smartAccount.latestAuthenticatorId;
 }
 
+/**
+ * Get the last authenticator id of the abstract account
+ * @param abstractAccount
+ * @returns
+ */
+export async function getAuthenticatorIdByAuthenticatorIndex(
+  abstractAccount: string,
+  authenticatorIndex: number,
+  indexerUrl: string,
+): Promise<number> {
+  const apolloClient = getApolloClient(indexerUrl);
+  const { data } = await apolloClient.query<{
+    smartAccounts: {
+      nodes: {
+        authenticators: {
+          nodes: {
+            authenticator: string;
+            authenticatorIndex: number;
+            id: string;
+            type: string;
+            version: string;
+          }[];
+        };
+        id: string;
+      }[];
+    };
+  }>({
+    query: SmartWalletIndexQueryByAccountId,
+    variables: {
+      id: abstractAccount,
+      index: authenticatorIndex,
+    },
+  });
+  if (!data || !data.smartAccounts) {
+    return 0;
+  }
+
+  // Always returning the first one found because this query should only return an array of 1
+  return (
```

Comment:
can we log when this isn't 1, since it is unexpected behavior

Reply:
I might actually be able to change the query so that it doesn't return an array, but if I can't then I'll just log this out